ZipBuilder/Sources/ZipBuilder/CarthageUtils.swift

```diff
@@ -51,23 +106,17 @@ extension CarthageUtils {
 
       // Parse the JSON file, ensure that we're not trying to overwrite a release.
       var jsonManifest = parseJSONFile(fromDir: jsonDir, product: product)
-      guard jsonManifest[firebaseVersion] == nil else {
-        print("Carthage release for \(product) \(firebaseVersion) already exists - skipping.")
-        continue
-      }
-
-      // Find all the .frameworks in this directory.
-      let allContents: [String]
-      do {
-        allContents = try FileManager.default.contentsOfDirectory(atPath: fullPath.path)
-      } catch {
-        fatalError("Could not get contents of \(product) for Carthage build in order to add " +
-          "an Info.plist in each framework. \(error)")
+      if !args.carthageSkipVersionCheck {
+        guard jsonManifest[firebaseVersion] == nil else {
+          print("Carthage release for \(product) \(firebaseVersion) already exists - skipping.")
+          continue
+        }
       }
 
-      // Carthage will fail to install a framework if it doesn't have an Info.plist, even though
-      // they're not used for static frameworks. Generate one and write it to each framework.
-      let frameworks = allContents.filter { $0.hasSuffix(".framework") }
+      // Make updates to all frameworks to make Carthage happy. We don't worry about xcframeworks
+      // here.
+      let allFiles = FileManager.default.enumerator(atPath: fullPath.path)?.allObjects as! [String]
```

Comment:
Force cast `as!` will crash with an unclear error even if `.enumerator(atPath: fullPath.path)` is `nil`. I think, we should use `guard` here.